src/main/java/core/basesyntax/CsvDataParser.java

```
@@ -0,0 +1,84 @@
+        }
+
+        return data.stream()
+                .skip(1)
+                .map(row -> {
+                    String[] values = getValues(row);
+
+                    Operation operation = this.calculateOperation(values[OPERATION_INDEX].trim());
+
+                    if (operation == null) {
+                        throw new RuntimeException("Operation is invalid "
+                                + row);
+                    }
+
+                    String fruitName = values[FRUIT_NAME_INDEX].trim();
+
+                    if (fruitName.isEmpty()) {
+                        throw new RuntimeException("Fruit name is missing"
+                                + row);
+                    }
+
+                    int quantity = this.getQuantity(values[QUANTITY_INDEX]);
+
+                    return new FruitTransaction(
+                            operation,
+                            fruitName,
+                            quantity
+                    );
+                })
+                .collect(Collectors.toList());
+    }
+
+    private String[] getValues(String data) {
+        String[] values = data.split(",");
+
+        if (values.length != 3) {
+            throw new RuntimeException("Wrong input format"
+                    + data);
+        }
+
+        return values;
+    }
+
+    private Operation calculateOperation(String operation) {
+        return Arrays.stream(Operation.values())
+                .filter(el -> el.getCode().equals(operation))
+                .findFirst()
+                .orElse(null);
+    }
+
+    private int getQuantity(String data) {
+        String quantity = data.trim();
+
+        if (quantity.isEmpty()) {
+            throw new RuntimeException("Quantity is empty "
+                    + quantity);
+        }
+
+        try {
+            return Integer.parseInt(quantity);
+        } catch (NumberFormatException e) {
+            throw new RuntimeException("Wrong input number "
+                    + data);
```

This violates checklist item #6: "Don't ignore exceptions in try-catch blocks". The original `NumberFormatException e` should be passed as the cause when creating the new `RuntimeException`. This provides more context for debugging, for example: `throw new RuntimeException("Wrong input number " + data, e);`.